src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificatePal.Windows.PrivateKey.cs

```diff
@@ -188,8 +194,51 @@ public ICertificatePal CopyWithPrivateKey(ECDiffieHellman ecdh)
 
         public ICertificatePal CopyWithPrivateKey(MLDsa privateKey)
         {
-            throw new PlatformNotSupportedException(
-                SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(MLDsa)));
+            if (privateKey is MLDsaCng mldsaCng)
+            {
+                CngKey key = mldsaCng.GetCngKey();
+
+                ICertificatePal? clone = CopyWithPersistedCngKey(key);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+
+                // TODO why are the other keys (ECDsa, Dsa, etc.) copying via export/import
+                // instead of doing the following? Maybe this fails if key isn't exportable
+                // (the export/import method can handle it by exporting encrypted pkcs#8)?
+                using (MLDsaCng clonedKey = new MLDsaCng(key))
+                {
+                    return CopyWithEphemeralKey(clonedKey.GetCngKey());
+                }
+            }
+
+            if (privateKey is MLDsaImplementation mldsaImplementation)
+            {
+                using (CngKey clonedKey = mldsaImplementation.CreateEphemeralCng())
+                {
+                    return CopyWithEphemeralKey(clonedKey);
+                }
+            }
+
+            // MLDsa other: either export pkcs#8 or do the dance with export seed -> export secret key
+            // Note for pkcs#8, implementors may give us only a private key even when there's a seed, but
+            // we don't care.
+            byte[] exportedPkcs8 = privateKey.ExportPkcs8PrivateKey();
```

Comment:
This pulled private key material into the managed heap (assuming it wasn't already).

So it needs to CryptographicOperations.ZeroMemory it after it's done reading from it.  (Which is what all the `using (PinAndClear.Track(privateParameters.D!))` is in CopyWithPrivateKey(RSA))

Reply:
That is the concern exactly, yep.